packages/edit-site/src/hooks/push-changes-to-global-styles/index.js

```diff
@@ -152,12 +228,22 @@ function PushChangesToGlobalStylesControl( {
 			set( newUserConfig, [ 'styles', 'blocks', name, ...path ], value );
 		}
 
+		const newBlockAttributes = {
+			borderColor: undefined,
+			backgroundColor: undefined,
+			textColor: undefined,
+			gradient: undefined,
+			fontSize: undefined,
+			fontFamily: undefined,
+			style: cleanEmptyObject( newBlockStyles ),
```

Comment:
Only clearing the individual border side properties would still leave empty objects in the block style attribute without this `cleanEmptyObject`.